Test failure in CI build 5364

The following test appears to have failed:

[#5364](https://circleci.com/gh/cockroachdb/cockroach/5364):

```
I0803 21:30:56.697490 491 storage/queue.go:290  processing range range=2 ("b"-"\xff\xff") from rangeGC queue...
I0803 21:30:56.697663 491 storage/range_gc_queue.go:114  destroying local data from range 2
W0803 21:30:56.698316 491 rpc/server.go:351  rpc: server cannot decode request: read tcp 127.0.0.1:45704: use of closed network connection
W0803 21:30:56.698412 491 rpc/server.go:351  rpc: server cannot decode request: read tcp 127.0.0.1:52310: use of closed network connection
W0803 21:30:56.698503 491 rpc/server.go:351  rpc: server cannot decode request: read tcp 127.0.0.1:59889: use of closed network connection
--- FAIL: TestRaftAfterRemoveRange (0.18s)
	client_raft_test.go:848: storage/store.go:1135: couldn't find range in rangesByKey btree
=== RUN TestRangeGCQueueDropReplica
I0803 21:30:56.700518 491 storage/engine/rocksdb.go:99  opening in-memory rocksdb instance
W0803 21:30:56.701213 491 rpc/tls.go:41  listening via tcp to 127.0.0.1:0 without TLS
I0803 21:30:56.701485 491 multiraft/multiraft.go:454  node 100000001 starting
I0803 21:30:56.701572 491 storage/range.go:988  gossiping cluster id  from store 1, range 1
I0803 21:30:56.701776 491 raft/raft.go:390  100000001 became follower at term 5
I0803 21:30:56.701831 491 raft/raft.go:207  newRaft 100000001 [peers: [100000001], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I0803 21:30:56.701882 491 raft/raft.go:464  100000001 is starting a new election at term 5
I0803 21:30:56.701898 491 raft/raft.go:403  100000001 became candidate at term 6
--
I0803 21:30:58.333523 491 raft/raft.go:426  100000001 became leader at term 6
I0803 21:30:58.334667 491 storage/range_command.go:821  range 64: new leader lease replica 1:1 00:00:00.000 +1.000s
--- PASS: TestStoreRangeManySplits (1.00s)
=== RUN: Example_rebalancing
--- PASS: Example_rebalancing (0.05s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	7.724s
=== RUN TestBatchBasics
I0803 21:30:49.267357 481 storage/engine/rocksdb.go:99  opening in-memory rocksdb instance
I0803 21:30:49.269133 481 storage/engine/rocksdb.go:124  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN TestBatchGet
I0803 21:30:49.271647 481 storage/engine/rocksdb.go:99  opening in-memory rocksdb instance
I0803 21:30:49.271927 481 storage/engine/rocksdb.go:124  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.00s)
=== RUN TestBatchMerge
I0803 21:30:49.273933 481 storage/engine/rocksdb.go:99  opening in-memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -830,23 +830,16 @@ func TestRaftAfterRemoveRange(t *testing.T) {
 	mtc.unreplicateRange(raftID, 0, 2)
 	mtc.unreplicateRange(raftID, 0, 1)
 
-	rng, err := mtc.stores[1].GetRange(raftID)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	// If the range removal happens before the range applies the replica config change, the group
-	// will be re-created when MultiRaft receives a MsgApp.
-	if err := util.IsTrueWithin(func() bool {
-		return len(rng.Desc().Replicas) == 1
-	}, 1*time.Second); err != nil {
-		t.Fatal(err)
-	}
-
-	// Remove the range from the second Store.
-	if err := mtc.stores[1].RemoveRange(rng); err != nil {
-		t.Fatal(err)
-	}
+	// Wait for the removal to be processed.
+	util.SucceedsWithin(t, time.Second, func() error {
+		_, err := mtc.stores[1].GetRange(raftID)
+		if _, ok := err.(*proto.RangeNotFoundError); ok {
+			return nil
+		} else if err != nil {
+			return err
+		}
+		return util.Errorf("range still exists")
+	})
 
 	if err := mtc.transport.Send(&multiraft.RaftMessageRequest{
 		GroupID: proto.RaftID(0),

